browser/browsing_data/brave_browsing_data_remover_delegate.cc

```diff
@@ -170,9 +172,11 @@ void BraveBrowsingDataRemoverDelegate::ClearIPFSCache() {
   }
 
   base::ThreadPool::PostTaskAndReply(
-      FROM_HERE, {base::TaskPriority::USER_VISIBLE, base::MayBlock()},
+      FROM_HERE,
+      {base::TaskPriority::USER_VISIBLE, base::MayBlock(),
+       base::TaskShutdownBehavior::CONTINUE_ON_SHUTDOWN},
       base::BindOnce(&BraveBrowsingDataRemoverDelegate::WaitForIPFSRepoGC,
-                     base::Unretained(this), base::Passed(&process)),
+                     weak_ptr_factory_.GetWeakPtr(), base::Passed(&process)),
```

Comment:
Can't use free function as suggested because we need to be a friend class of ScopedAllowBaseSyncPrimitives in order to instantiate it. See base/threading/thread_restrictions.h.

Reply:
free function can also be a friend, see `PathProviderWin` in that header

Reply:
Ah, thanks for pointing that out, I wasn't aware of that. :)
Would it be okay to merge as is or do you think we should definitely use free function here instead?

Reply:
up to you! free function will just save a couple of lines

Reply:
k, I'll merge as is then, thanks for the review!